frontend/src/contexts/Paging.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+/* Pagination */
+type IPage = '/' | '/bp';
+
+const PageContext = React.createContext<{ page: IPage; setPage(p: IPage): void }>({ page: '/', setPage() { } });
+
+export const WithPagination = ({ children }: React.PropsWithChildren<{}>) => {
+    const [page, _setPage] = React.useState<IPage>(window.location.pathname as IPage);
```

Comment:
```suggestion
    const [page, _setPage] = useState(window.location.pathname as IPage);
```

Reply:
Is there a reason why `_setPage` is prefixed with `_`?

Reply:
Cause `setPage` gets reimplemented later on. That's my personal coding style though ... I'm fine with another name

Reply:
Ok... then maybe leave a comment to make this clear.

Reply:
I guess as we only have two days we should not focus on code quality *that* much